internal/blockchain/ethereum/ffi_param_validator.go

```diff
@@ -33,31 +33,40 @@ var bytesRegex, _ = regexp.Compile("^bytes([0-9]{1,2})?")
 func (v *FFIParamValidator) Compile(ctx jsonschema.CompilerContext, m map[string]interface{}) (jsonschema.ExtSchema, error) {
```

Comment:
I was trying to work out where this is used. I think we agreed that the connector would be responsible for verifying input, and that we should ensure that happens synchronously before we give back a `202` to an invoke.

Understand this PR might not be the final step in the chain, but I want to make sure we're not trying to replicate the connector input validation logic in core.

Reply:
The most important use of this is in the contracts manager https://github.com/hyperledger/firefly/blob/e8419846ccd8c6e80791540a06881f3385ace2da/internal/contracts/manager.go#L112-L118

Once this JSON Schema extension is registered, the JSON Schema library calls this function when a request is made to broadcast a new FFI. This ensures that the schema specified in the FFI for each field adheres to the specific rules defined by each blockchain connector. As an example, the Ethereum connector requires the `details` object in most places, and it requires that a `type` field is set inside there (in addition to a bunch of other schema rules).

In other words, this is the metaschema (the schema for the schema). This code is _not_ for validating blockchain invocation request input.